main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -2192,9 +2192,11 @@ object Resolver {
   private def semiResolveType(tpe0: NamedAst.Type, wildness: Wildness, env: ListMap[String, Resolution], ns0: Name.NName, root: NamedAst.Root)(implicit level: Level, flix: Flix): Validation[UnkindedType, ResolutionError] = {
     def visit(tpe0: NamedAst.Type): Validation[UnkindedType, ResolutionError] = tpe0 match {
       case NamedAst.Type.Var(ident, loc) =>
-        val symVal = lookupTypeVar(ident, wildness, env)
-        mapN(symVal) {
-          case sym => UnkindedType.Var(sym, loc)
+        lookupTypeVar(ident, wildness, env) match {
+          case Result.Ok(sym) => UnkindedType.Var(sym, loc).toSuccess
+          case Result.Err(e) =>
+            // Note: We assume the default type variable has kind Star.
```

Comment:
See other PR; We should have `UnkindedType.Error`

Reply:
Yeah, but does it solve it or?

Reply:
What I mean is; will we be able to kind it correctly?

Reply:
It should eliminate some spurious kind errors